jina/peapods/peas/jinad.py

```diff
@@ -0,0 +1,255 @@
+import copy
+import asyncio
+import argparse
+import threading
+import multiprocessing
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Union
+
+from . import BasePea
+from .helper import _get_worker
+from ...helper import get_or_reuse_loop
+from ...jaml.helper import complete_path
+from ...importer import ImportExtensions
+from ...enums import replace_enum_to_str
+from ...logging.logger import JinaLogger
+from ...excepts import (
+    DaemonConnectivityError,
+    DaemonPeaCreationFailed,
+    DaemonWorkspaceCreationFailed,
+)
+
+if TYPE_CHECKING:
+    import argparse
+
+
+class JinaDProcessTarget:
+    """Target to be executed on JinaD Process"""
+
+    def __call__(
+        self,
+        args: 'argparse.Namespace',
+        is_started: Union['multiprocessing.Event', 'threading.Event'],
+        is_shutdown: Union['multiprocessing.Event', 'threading.Event'],
+        is_ready: Union['multiprocessing.Event', 'threading.Event'],
+        is_cancelled: Union['multiprocessing.Event', 'threading.Event'],
+    ):
+        """Method responsible to manage a remote Pea
+
+        This method is the target for the Pea's `thread` or `process`
+
+        .. note::
+            Please note that env variables are process-specific. Subprocess inherits envs from
+            the main process. But Subprocess's envs do NOT affect the main process. It does NOT
+            mess up user local system envs.
+
+        :param args: namespace args from the Pea
+        :param is_started: concurrency event to communicate runtime is properly started. Used for better logging
+        :param is_shutdown: concurrency event to communicate runtime is terminated
+        :param is_ready: concurrency event to communicate runtime is ready to receive messages
+        :param is_cancelled: concurrency event to receive cancelling signal from the Pea. Needed by some runtimes
+        """
+        self.args = args
```

Comment:
Make sure to handle terminate signal properly so that terminate is captured and handled properly.